tools/tpm2_nvread.c

```diff
@@ -259,14 +277,46 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
         { "offset"      , required_argument, NULL, 'o' },
         { "handle-passwd", required_argument, NULL, 'P' },
         { "input-session-handle",1,          NULL, 'S' },
+        {"set-list",       required_argument, NULL, 'L' },
+        {"pcr-input-file", required_argument, NULL, 'F' },
     };
 
-    *opts = tpm2_options_new("x:a:f:s:o:P:S:", ARRAY_LEN(topts), topts, on_option, NULL);
+    *opts = tpm2_options_new("x:a:f:s:o:P:S:L:F:", ARRAY_LEN(topts),
+            topts, on_option, NULL);
 
     return *opts != NULL;
 }
 
 int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
 
-    return nv_read(sapi_context, flags) != true;
+    if (ctx.flags.L && ctx.flags.F) {
```

Comment:
...but here is checking for both, so it makes -F to be a required option. Instead, it should only check for ctx.flags.L since tpm2_policy_build() can take a NULL argument for the raw_pcrs_file parameter. If is NULL, then  Tss2_Sys_PCR_Read() will be called for the PCR selection.

Reply:
@williamcroberts I noticed that the tpm2_unseal tool had the same bug, so I fixed on PR #595.

Reply:
Ok sounds good, I literally copied you!